source/jormungandr/jormungandr/transient_socket.py

```diff
@@ -61,107 +60,89 @@ class TransientSocket(object):
 
     """
 
-    # TODO: use dataclass in python3.7
-    _Socket = namedtuple('_Socket', ['t', 'socket'])
+    # TODO: use dataclass in python > 3.7
+    TimedSocket = namedtuple('TimedSocket', ['t', 'socket'])
 
     # _sockets is a map of TransientSocket vs a sorted list of tuple of created time and tcp sockets
     # the sorted list is arranged in a way that the first element is the most recent one and the last element is oldest
     # one.
-    _sockets = defaultdict(lambda: SortedList([], key=lambda s: -s.t))  # type: Dict[TransientSocket, SortedList]
-    _reaper_interval = 10
     _logger = logging.getLogger(__name__)
 
-    def __init__(self, name, zmq_context, socket_path, socket_ttl, *args, **kwargs):
+    def __init__(self, name, zmq_context, zmq_socket, socket_ttl, *args, **kwargs):
         super(TransientSocket, self).__init__(*args, **kwargs)
         self.name = name
         self._zmq_context = zmq_context
-        self._socket_path = socket_path
+        self._zmq_socket = zmq_socket
         self.ttl = socket_ttl
+        self.semaphore = BoundedSemaphore(1)
+        self._sockets = SortedList([], key=lambda s: -s.t)
+
+    def make_new_socket(self):
+        start = time.time()
+        socket = self._zmq_context.socket(zmq.REQ)
+        socket.connect(self._zmq_socket)
+        self._logger.info(
```

Comment:
```suggestion
        self._logger.debug(
```
info is a bit too verbose for this kind of logs no ?